### PR TITLE
Managed buffers additional hooks

### PR DESCRIPTION
**Pull request by andrei-marinica**
- append bytes vs. append another managed buffer entirely
- slice -> wasm memory
- slice -> another managed buffer

### CODE REVIEW

File: arwen/elrondapi/manBufOps.go [hunk 1]
```diff
@@ -141,75 +164,163 @@ func v1_4_mBufferNewFromBytes(context unsafe.Pointer, dataOffset int32, dataLeng
...
 
-	gasToUse := metering.GasSchedule().ManagedBufferAPICost.MBufferSetBytes
+	gasToUse := metering.GasSchedule().ManagedBufferAPICost.MBufferGetLength
 	metering.UseGas(gasToUse)
-	managedType.ConsumeGasForThisIntNumberOfBytes(int(dataLength))
 
-	data, err := runtime.MemLoad(dataOffset, dataLength)
+	length := managedType.GetLength(mBufferHandle)
+	if length == -1 {
+		arwen.WithFault(arwen.ErrNoManagedBufferUnderThisHandle, context, runtime.ManagedBufferAPIErrorShouldFailExecution())
```
Comment (sasurobert) [inline arwen/elrondapi/manBufOps.go:178]: _ = arwen.WithFault.

File: arwen/elrondapi/manBufOps.go [hunk 2]
```diff
@@ -141,75 +164,163 @@ func v1_4_mBufferNewFromBytes(context unsafe.Pointer, dataOffset int32, dataLeng
...
+	metering := arwen.GetMeteringContext(context)
+
+	gasToUse := metering.GasSchedule().ManagedBufferAPICost.MBufferSetBytes
+	metering.UseGas(gasToUse)
+
+	dataBufferBytes, err := managedType.GetBytes(dataHandle)
+	if arwen.WithFault(err, context, runtime.ManagedBufferAPIErrorShouldFailExecution()) {
+		return 1
+	}
+	managedType.ConsumeGasForThisIntNumberOfBytes(len(dataBufferBytes))
```
Comment (sasurobert) [inline arwen/elrondapi/manBufOps.go:308]: append should cost more than a simple setbytes. as there is at least one operation in plus

**Review by sasurobert — CHANGES_REQUESTED** [review]

File: test/features/basic-features/mandos/managed_buffer_concat_1.scen.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                "function": "mbuffer_concat_1",
+                "arguments": [
+                    "str:abc",
+                    ""
+                ]
+            },
+            "expect": {
+                "out": [ "str:abc" ]
+            }
+        }
```
Comment (andrei-diaconescu) [inline test/features/basic-features/mandos/managed_buffer_concat_1.scen.json:64]: optional: add step with 2 empty buffers?

File: arwen/elrondapi/manBufOps.go (hunk 2 above)
Comment (andrei-marinica) [reply arwen/elrondapi/manBufOps.go:308]: we have below:
```
       gasToUse = math.MulUint64(metering.GasSchedule().BaseOperationCost.DataCopyPerByte, uint64(len(data)))
	metering.UseGas(gasToUse)
```
Do we need more than that?

Comment (andrei-diaconescu) [reply arwen/elrondapi/manBufOps.go:308]: i think the base gasCost should be MBufferSetBytes + MBufferGetBytes, than use the extra DataCopyPerByte

**Review by andrei-diaconescu — APPROVED** [review]

**Review by andrei-diaconescu — APPROVED** [review]

File: arwenmandos/gasSchedules/tomlfiles.go [hunk 4]
```diff
@@ -807,32 +775,6 @@ const (
     VerifyBLS              = 5000000
     VerifyEd25519          = 2000000
     VerifySecp256k1        = 2000000
```
Comment (sasurobert) [inline arwenmandos/gasSchedules/tomlfiles.go:777]: do not delete these. why did you delete ?

**Review by sasurobert — APPROVED** [review]

File: arwenmandos/gasSchedules/tomlfiles.go (hunk 4 above)
Comment (andrei-diaconescu) [reply arwenmandos/gasSchedules/tomlfiles.go:777]: they were not necessary. only necessary in gasSchedule v3 I think
